scripts/masterCron.ts

```diff
@@ -86,6 +88,42 @@ const getNextFriday = (currentExpiry: number) => {
   return friday8am;
 };
 
+function generateTokenSet(tokens: Array<object>) {
+  // reference: https://github.com/opynfinance/opyn-tokenlist/blob/master/opyn-v1.tokenlist.json
+  let tokenJSON = {
+    name: "Ribbon oTokens",
+    logoURI: "https://i.imgur.com/u5z1Ev2.png",
+    keywords: ["defi", "option", "opyn", "ribbon"],
+    //convert to something like 2021-09-08T10:51:49Z
+    timestamp: moment.format().toString().slice(0, -6) + "Z",
+    version: { major: 1, minor: 0, patch: 0 },
+    tokens: tokens,
+  };
+
+  return JSON.stringify(tokenJSON);
+}
+
+async function pushTokenListToGit(tokenlist: string, fileName: string) {
+  const options: Partial<SimpleGitOptions> = {
+    baseDir: process.cwd(),
+    binary: "git",
+    maxConcurrentProcesses: 6,
+  };
+
+  // when setting all options in a single object
+  const git: SimpleGit = simpleGit(options);
+
+  fs.writeFile(`../ribbon-token-list/${fileName}`, tokenlist);
+
+  await git
+    .cwd("../ribbon-token-list")
+    .addConfig("user.name", "cron job")
+    .addConfig("user.email", "some@one.com")
```

Comment:
Need to add actual credentials through .env

Reply:
this what you mean or is there something else

https://github.com/steveukx/git-js#authentication

Reply:
Yep. You can pass in the username and password to the function. But the credentials need to be stored in .env

Reply:
actually don't need to clone each time. can just clone once and set up the personal access token for the bot account for pushing. _should_ just be able to push in the script without any extra config right.